src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(...nums) {
+    if (nums.length === 0) {
+      const resultSum = sum;
+
+      sum = 0;
+
+      return resultSum;
+    }
+
+    sum += nums.reduce((total, elem) => total + elem);
```

Comment:
The initial value of 0 should be provided as the second parameter to the 'reduce' function to handle cases when the 'nums' array is empty.